src/Symfony/Component/Intl/Data/Generator/LocaleDataGenerator.php

```diff
@@ -83,19 +83,14 @@ protected function preGenerate()
      */
     protected function generateDataForLocale(BundleEntryReaderInterface $reader, $tempDir, $displayLocale)
     {
-        // Generate aliases, needed to enable proper fallback from alias to its
-        // target
+        // Don't generate aliases, as they are resolved during runtime
         if (isset($this->localeAliases[$displayLocale])) {
-            return ['%%ALIAS' => $this->localeAliases[$displayLocale]];
+            return;
         }
 
         // Generate locale names for all locales that have translations in
         // at least the language or the region bundle
-        try {
-            $displayFormat = $reader->readEntry($tempDir.'/lang', $displayLocale, ['localeDisplayPattern']);
-        } catch (MissingResourceException $e) {
-            $displayFormat = $reader->readEntry($tempDir.'/lang', 'root', ['localeDisplayPattern']);
-        }
+        $displayFormat = $reader->readEntry($tempDir.'/lang', $displayLocale, ['localeDisplayPattern']);
```

Comment:
defaulting to root during compile makes things like this more straightforward (the timezone generator in master also benefits)

i've verified it causes no side-effects in 4.2 / master :+1: